pkg/sql/catalog/typedesc/type_desc.go

```diff
@@ -796,6 +796,9 @@ func EnsureTypeIsHydrated(
 				return err
 			}
 		}
+	} else if t.Family() == types.ArrayFamily {
```

Comment:
based on the `UserDefinedArrayOID` func, i would have expected the logic below to have been run for arrays too.

should we call `EnsureTypeIsHydrated(ctx, t.ArrayContents(), res)` but only return early if `err != nil`?